bundles/core/org.eclipse.smarthome.core.persistence/src/main/java/org/eclipse/smarthome/core/persistence/internal/rest/PersistenceResource.java

```diff
@@ -0,0 +1,237 @@
+
+    static private Map<String, PersistenceService> persistenceServices = new HashMap<String, PersistenceService>();
+
+    public void addPersistenceService(PersistenceService service) {
+        persistenceServices.put(service.getName(), service);
+    }
+
+    public void removePersistenceService(PersistenceService service) {
+        persistenceServices.remove(service.getName());
+    }
+
+    @GET
+    @Produces({ MediaType.APPLICATION_JSON })
+    @ApiOperation(value = "Gets a list of persistence services.", response = String.class, responseContainer = "List")
+    @ApiResponses(value = @ApiResponse(code = 200, message = "OK") )
+    public Response httpGetPersistenceServices(@Context HttpHeaders headers) {
+        Object responseObject = getPersistenceServiceList();
+        return Response.ok(responseObject).build();
+    }
+
+    @GET
+    @Path("/{itemname: [a-zA-Z_0-9]*}")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @ApiOperation(value = "Gets item persistence data from the persistence service.", response = ItemHistoryBean.class)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 404, message = "Unknown Item or persistence service") })
+    public Response httpGetPersistenceItemData(@Context HttpHeaders headers,
+            @ApiParam(value = "The item name", required = true) @PathParam("itemname") String itemName,
+            @ApiParam(value = "Name of the persistence service. If not provided the default service will be used", required = false) @QueryParam("servicename") String serviceName,
+            @ApiParam(value = "Start time of the data to return. Will default to 1 day before endtime", required = false) @QueryParam("starttime") String startTime,
+            @ApiParam(value = "End time of the data to return. Will default to current time.", required = false) @QueryParam("endtime") String endTime,
+            @ApiParam(value = "Page number of data to return. This parameter will enable paging.", required = false) @QueryParam("page") int pageNumber,
+            @ApiParam(value = "The length of each page.", required = false) @QueryParam("pagelength") int pageLength) {
+
+        final Object responseObject = getItemHistoryBean(serviceName, itemName, startTime, endTime, pageNumber,
+                pageLength);
+        return Response.ok(responseObject).build();
+    }
+
+    private Date convertTime(String sTime) {
+        DateTimeType dateTime = new DateTimeType(sTime);
+        return dateTime.getCalendar().getTime();
+    }
+
+    private Response getItemHistoryBean(String serviceName, String itemName, String timeBegin, String timeEnd,
+            int pageNumber, int pageLength) {
+        // Benchmarking timer...
+        long timerStart = System.currentTimeMillis();
+
+        // If serviceName is null, then use the default service
+        PersistenceService service = null;
+        if (serviceName == null) {
```

Comment:
> Feel free to leave that out of this PR and I will see if I can come up with a good solution.

Perhaps we should move the comment (in the code) to the branch and change the comment that it does not state that a default is used ATM.

```diff
+        PersistenceService service = null;
+        if (serviceName == null) {
+            // Add handing (e.g. use default persistence service) here some time.
 +        } else {
+            service = persistenceServices.get(serviceName);
+        }
```